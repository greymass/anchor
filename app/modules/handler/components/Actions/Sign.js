// @flow
import React, { Component } from 'react';
import { withTranslation } from 'react-i18next';
import { Button, Header, Icon, Label } from 'semantic-ui-react';

class PromptActionSign extends Component<Props> {
  render() {
    const {
      broadcast,
      disabled,
      loading,
      onClick,
      t,
      wallet,
    } = this.props;
    return (
      <Button
        animated="fade"
        as="div"
        disabled={disabled}
        floated="right"
        labelPosition="left"
        onClick={() => onClick(broadcast)}
      >
        <Label
          as="a"
          basic
          pointing="right"
        >
          <Header textAlign="left">
            <Header.Content>
              <Header.Subheader style={{ fontWeight: 'bold' }}>
                {t('handler_action_sign_header')}
              </Header.Subheader>
              {wallet.account}@{wallet.authorization}
            </Header.Content>
          </Header>
        </Label>
        <Button
          color="green"
          icon
          loading={loading}
          style={{ position: 'relative' }}
        >
          <Button.Content hidden style={{ marginTop: '-0.75em' }}>
            <Icon
              name="signup"
              size="large"
              style={{ marginRight: 0 }}
            />
          </Button.Content>
          <Button.Content visible style={{ margin: '0 1em' }}>
            <Icon
              name="pencil"
              size="large"
              style={{ marginRight: 0 }}
            />
          </Button.Content>
        </Button>
      </Button>
    );
  }
}

export default withTranslation('handler')(PromptActionSign);
