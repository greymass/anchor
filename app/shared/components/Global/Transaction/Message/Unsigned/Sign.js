// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Button, Container, Divider, Header, Segment } from 'semantic-ui-react';

import GlobalTransactionViewActions from '../../../Transaction/View/Actions';
import GlobalTransactionViewDetail from '../../../Transaction/View/Detail';
import GlobalTransactionViewFull from '../../../Transaction/View/Full';

class GlobalTransactionMessageUnsignedSign extends Component<Props> {
  signbroadcast = () => {
    const {
      actions,
      contract,
      transaction
    } = this.props;
    const {
      signTransaction
    } = actions;
    signTransaction(transaction, contract);
  }
  render() {
    const {
      onClose,
      t,
      transaction
    } = this.props;

    let actions = [];
    if (transaction && transaction.transaction) {
      ({ actions } = transaction.transaction.transaction);
    }
    const signed = (transaction.transaction.signatures.length > 0);
    return (
      <Segment basic>
        {(transaction)
          ? (
            <React.Fragment>
              <Header>
                {t('broadcast_transaction_confirm_details_title')}
                <Header.Subheader>
                  {t('broadcast_transaction_confirm_details_content')}
                </Header.Subheader>
              </Header>
              <GlobalTransactionViewDetail
                broadcastable={!signed}
                transaction={{data: transaction}}
              />
              <Container textAlign="center">
                <Button
                  color="purple"
                  content={t('broadcast_transaction_signbroadcast')}
                  icon="wifi"
                  onClick={this.signbroadcast}
                  size="large"
                />
                <Button
                  color="grey"
                  content={t('close')}
                  icon="x"
                  onClick={onClose}
                  size="large"
                />
              </Container>
              <Segment>
                <GlobalTransactionViewActions
                  actions={actions}
                />
                <Divider />
                <GlobalTransactionViewFull
                  transaction={transaction}
                />
              </Segment>
            </React.Fragment>
          )
          : (
            <Header>
              {t('broadcast_transaction_no_details_title')}
              <Header.Subheader>
                {t('broadcast_transaction_no_details_content')}
              </Header.Subheader>
            </Header>
          )
        }
      </Segment>
    );
  }
}

export default translate('transaction')(GlobalTransactionMessageUnsignedSign);
