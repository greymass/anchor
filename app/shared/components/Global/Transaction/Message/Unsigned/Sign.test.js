import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalTransactionMessageUnsignedSign } from './Sign';

const mockProps = {
  onClose: () => null,
  t: text => text,
  transaction: {
    transaction: {
      signatures: [],
      transaction: {
        actions: []
      }
    }
  },
};

describe('GlobalTransactionMessageUnsignedSign', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalTransactionMessageUnsignedSign {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

