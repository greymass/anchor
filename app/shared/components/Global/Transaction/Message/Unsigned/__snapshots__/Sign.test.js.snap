// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTransactionMessageUnsignedSign renders correctly 1`] = `
<Segment
  basic={true}
>
  <Header>
    broadcast_transaction_confirm_details_title
    <HeaderSubheader>
      broadcast_transaction_confirm_details_content
    </HeaderSubheader>
  </Header>
  <Translate(GlobalTransactionViewDetail)
    broadcastable={true}
    transaction={
      Object {
        "data": Object {
          "transaction": Object {
            "signatures": Array [],
            "transaction": Object {
              "actions": Array [],
            },
          },
        },
      }
    }
  />
  <Container>
    <Button
      as="button"
      color="purple"
      content="broadcast_transaction_signbroadcast"
      floated="right"
      icon="wifi"
      onClick={[Function]}
      size="large"
    />
    <Button
      as="button"
      color="grey"
      content="close"
      icon="x"
      onClick={[Function]}
      size="large"
    />
  </Container>
  <Segment>
    <Translate(GlobalTransactionViewActions)
      actions={Array []}
    />
    <Divider />
    <Translate(GlobalTransactionViewFull)
      transaction={
        Object {
          "transaction": Object {
            "signatures": Array [],
            "transaction": Object {
              "actions": Array [],
            },
          },
        }
      }
    />
  </Segment>
</Segment>
`;
